src/app/hero-details/hero-details.component.html

```diff
@@ -0,0 +1,27 @@
+<h3>Hero Name</h3>
+<div [formGroup]="form.controls.name">
+    <div class="form-group row">
+        <label class="col-sm-2">Alias: *</label>
+        <input class="col-sm-10 form-control" formControlName="alias" [class.is-invalid]="form.controls.name.controls.alias.touched && form.controls.name.controls.alias.invalid" />
```

Comment:
Is there any better way to add the class to this `input` using `formControlName`? Additionally, this fails the AoT build because `form.controls.name` is an `AbstractControl` for TS which does not have the `.controls` property.

Reply:
I thought about I might be able to do something along the lines of `#alias="formControl"` so I am then able to access it like `alias.touched && alias.invalid` but it does not seem that `formControlName` has an `exportAs` somewhere.